src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -136,27 +136,19 @@ public interface INamedTypeSymbol : ITypeSymbol
         bool MightContainExtensionMethods { get; }
 
         /// <summary>
-        /// Returns the types of the elements for types that are tuples.
-        ///
-        /// If this type is not a tuple, then returns default.
-        /// </summary>
-        ImmutableArray<ITypeSymbol> TupleElementTypes { get; }
-
-        /// <summary>
-        /// Returns the friendly-names of the elements for types that are tuples and that have friendly-names.
-        ///
-        /// If this type is not a tuple, then returns default.
-        /// If this type has no friendly-names, then returns default.
-        /// </summary>
-        ImmutableArray<string> TupleElementNames { get; }
```

Comment:
`TypeSymbol.TupleElementTypes` and `TypeSymbol.TupleElementNames`  are useful internally for purposes such as generic instantiations 

---
In reply to: [86593243](https://github.com/dotnet/roslyn/pull/14958#discussion_r86593243) [](ancestors = 86593243)